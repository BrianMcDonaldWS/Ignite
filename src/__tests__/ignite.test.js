import path from 'path';
import {
  initOptions,
  initPlugins,
  initBlogPosts,
  initSearchIndex,
  initBuildMessages,
  getAuthor
} from '../ignite';

describe('initPlugins', () => {
  test('should initialize options', async () => {
    const result = await initPlugins({
      plugins: [['pluginName', '../extensions/notRealExtension/']]
    });
    expect(result).toEqual({
      plugins: [['pluginName', '../extensions/notRealExtension/', {}]]
    });
  });

  test('should call init function if present', async () => {
    const pluginPath = path.resolve('src/extensions/testExtension/');
    const result = await initPlugins({
      plugins: [['pluginName', pluginPath, { foo: 'bar' }]]
    });
    expect(result).toMatchSnapshot();
  });
});

describe('initBlogPosts', () => {
  test('do nothing with when docs contain no blog posts', async () => {
    const result = await initBlogPosts({ src: '/' });
    expect(result).toBeNull();
  });

  test('attach creation date to each blog post', async () => {
    const result = await initBlogPosts({
      src: 'examples/blog/docs'
    });
    expect(result).toEqual([
      { birth: 1526593244000, path: 'blog/FirstPost.md' },
      { birth: 1526623637000, path: 'blog/ShorterPost.md' }
    ]);
  });
});

test('getAuthor', () => {
  expect(getAuthor()).toEqual({
    email: 'andrew_lisowski@intuit.com',
    name: 'Andrew Lisowski'
  });
});

test('initSearchIndex', async () => {
  const result = await initSearchIndex({
    baseURL: '/',
    src: path.resolve('examples/multi-root/docs')
  });
  expect(result).toMatchSnapshot();
});

test('initBuildMessages', () => {
  const prod = initBuildMessages({});
  const dev = initBuildMessages({
    watch: true
  });

  expect(dev.compilationSuccessInfo).not.toEqual(prod.compilationSuccessInfo);
});

describe('initOptions', () => {
  test('uses rc file', async () => {
    const result = await initOptions({
      src: path.resolve('examples/multi-root/docs')
    });
    expect(result).toMatchSnapshot();
  });
});
