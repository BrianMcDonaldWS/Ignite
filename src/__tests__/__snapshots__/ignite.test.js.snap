// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initOptions uses rc file 1`] = `
Object {
  "baseURL": "/_ignite/",
  "blogPosts": null,
  "bulmaTheme": "default",
  "codeStyle": "foundation",
  "color": "auto",
  "compilationSuccessInfo": Object {
    "messages": Array [
      "Documentation built!",
    ],
    "notes": Array [
      "Bundled documentation stored in '_ignite/'.",
      "Run \`ignite --publish\` to publish documentation to github-pages.",
    ],
  },
  "dst": "_ignite/",
  "index": "index.md",
  "log": true,
  "logo": "logo.svg",
  "mode": "production",
  "navItems": Object {
    "First": "/_ignite/Root1",
    "Second": "/_ignite/Root2",
    "root": "/_ignite/Root1",
  },
  "open": true,
  "plugins": Array [],
  "port": 8008,
  "searchIndex": Array [
    Object {
      "body": "# Some Other Page
",
      "id": "Root1/Another.md",
    },
    Object {
      "body": "# First Page /.firstPage\\\\
",
      "id": "Root1/Page.md",
    },
    Object {
      "body": "- [First Page](/_ignite/Root1/Page.html)
- [Another Page](/_ignite/Root1/Another.html)
",
      "id": "Root1/index.md",
    },
    Object {
      "body": "# Second Page /.secondPage\\\\
",
      "id": "Root2/Page.md",
    },
    Object {
      "body": "- [Second Page](/_ignite/Root2/Page.html)
",
      "id": "Root2/index.md",
    },
  ],
  "selectedColor": "auto",
  "src": "examples/multi-root/docs",
  "static": false,
  "title": "Multi Root",
  "webpackPlugins": Array [],
}
`;

exports[`initPlugins should call init function if present 1`] = `
Object {
  "plugins": Array [
    Array [
      "pluginName",
      "src/extensions/testExtension/",
      Object {
        "_initData": Object {
          "data": "foo",
          "nested": Object {
            "deeply": Object {
              "func": [Function],
              "value": "foo",
            },
          },
        },
        "foo": "bar",
      },
    ],
  ],
}
`;

exports[`initSearchIndex 1`] = `
Array [
  Object {
    "body": "# Some Other Page
",
    "id": "Root1/Another.md",
  },
  Object {
    "body": "# First Page /.firstPage\\\\
",
    "id": "Root1/Page.md",
  },
  Object {
    "body": "- [First Page](/examples/multi-root/docs/Root1/Page.html)
- [Another Page](/examples/multi-root/docs/Root1/Another.html)
",
    "id": "Root1/index.md",
  },
  Object {
    "body": "# Second Page /.secondPage\\\\
",
    "id": "Root2/Page.md",
  },
  Object {
    "body": "- [Second Page](/examples/multi-root/docs/Root2/Page.html)
",
    "id": "Root2/index.md",
  },
]
`;
